src/include/daos_srv/pool.h

```diff
@@ -543,4 +544,17 @@ int ds_pool_svc_lookup_leader(uuid_t uuid, struct ds_pool_svc **ds_svcp, struct
 
 void ds_pool_svc_put_leader(struct ds_pool_svc *ds_svc);
 
+static inline bool
+ds_pool_rebuild_needed(struct ds_pool *pool)
+{
+	if (pool->sp_disable_rebuild)
+		return false;
+	if (!(pool->sp_self_heal & (DAOS_SELF_HEAL_AUTO_REBUILD | DAOS_SELF_HEAL_DELAY_REBUILD)))
+		return false;
+	if (pool->sp_reint_mode == DAOS_REINT_MODE_NO_DATA_SYNC)
+		return false;
```

Comment:
Does the reint mode also apply to rebuild jobs for exclude, drain, etc.?

Reply:
I removed the check for reint mode, i think it will be removed in the 2.8 as well.